### DISCUSSION TITLE
Listening to row level changes

### DISCUSSION THREAD

**Discussion by HassanIyan** [supabase/supabase#4204] [Questions]
a couple days ago listening to real time row level changes stop working for me... before that it works perfectly.

here is my code
```
const mySubscription = supabase
  .from('trips:id=eq.150')
  .on('UPDATE', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()
```

**Comment by GaryAustin1** [comment]
They are having various issues today with the new realtime with RLS drop, but not sure how that would impact you days ago.
You can do .subscribe((status)=>{console.log('subscribe',status);) to get some info on status of the connection (that also might show up in your network requests). 
Otherwise your code looks fine, so it is likely something on the realtime setup or table that changed or some code change causing your .subscribe not to run.

**Reply by GaryAustin1** [reply]
I just reported an issue on realtime behavior being different with the new changes for a non RLS table.  After about an hour my subscriptions (just like yours look) time out with error from then on.   I'm not sure exactly when it started for me but I don't think the day before yesterday.  I'm on Supabase online.

**Comment by w3b6x9** [comment]
@HassanIyan can you please email support@supabase.io with a link to this discussion and your project ref so we can investigate?

**Comment by Tarang** [comment]
I am having the same issue. There is nothing coming down the websocket anymore. Even if i authenticate I can do the `.from(xx).select()` commands but the subscription ones dont seem to respond.

Looking at the websocket wire as well it doesn't seem to authenticate.

**Reply by Tarang** [reply]
The code i used is

```
const supabaseUrl = 'https://**.supabase.co'
const supabaseKey = "<API_KEY>"
const supabase = createClient(supabaseUrl, supabaseKey)
const { user, error } = await supabase.auth.setAuth("<SIGNED JWT>")

//Works
console.log("AUTH", user, error);

const { data } = await supabase.from('d').select()

//WORKS!
console.log("D", data)

const d = supabase
  .from('data')
  .on('*', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()

console.log(d);
```

The websocket wire looks like:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/1616628/144361125-5a76e0c8-40a6-45b9-b295-a2829c275f52.png">

**Reply by w3b6x9** [reply]
@Tarang can you please email support@supabase.io with your project ref? I'll investigate.

**Comment by w3b6x9** [comment]
> a couple days ago listening to real time row level changes stop working for me... before that it works perfectly.
> 
> here is my code
> 
> ```
> const mySubscription = supabase
>   .from('trips:id=eq.150')
>   .on('UPDATE', payload => {
>     console.log('Change received!', payload)
>   })
>   .subscribe()
> ```

@HassanIyan just based on your code alone I wanted to point out that there's an existing issue with filtering on numeric columns. We're currently rolling out the fix to all projects, and depending on your project's region, your project may already have the fix.

**Comment by megacherry** [comment]
hi @w3b6x9 in my project realtime also stopped working. To test it i disabled RLS and deactivated/activated realtime for this table, but nothing happened. i doubted myself, should i write you an e-mail as well? :)

**Reply by w3b6x9** [reply]
@megacherry please do! Email is support@supabase.io and please include a link to your [comment](https://github.com/supabase/supabase/discussions/4204#discussioncomment-1748351) as well as your project ref and I can investigate.

**Comment by ferhatckrk** [comment]
hi, Although realtime listener worked fine about 1 week ago, when I test again today it does not work, I am working with flutter. What could be the reason for this? Replication settings on

**Reply by w3b6x9** [reply]
@thecavus have you upgraded to the latest `supabase-dart` version? Please send an email to support@supabase.io with your project ref and we'll investigate.

**Comment by w3b6x9** [comment]
Hello everyone, we're in the process of upgrading all Supabase projects to the newest Realtime RLS that addresses the issues mentioned in this Discussion. Can you please try again and see if your issues have been fixed?

**Reply by sidthekid402** [reply]
I still have this issue :(

**Reply by GaryAustin1** [reply]
@sidthekid402 This is a rather broad topic, what is your particular issue?  Do you have replication on for your table and is RLS set for selects properly.

**Reply by sidthekid402** [reply]
My apologies, supabase newbie here. I am trying to listen for changes on specific rows using the references here: https://supabase.com/docs/reference/javascript/subscribe#listening-to-row-level-changes . Any changes in the specific rows I'm listening to aren't detected as UPDATE's. The table allows read access for all users. Is RLS necessary for this? I don't believe I have RLS enabled.

**Reply by GaryAustin1** [reply]
In the UI goto database/replication make sure you have update toggled and then over in the far right you should see "table" click on that and make sure your table is checked.  RLS is not required, but you must meet the select policy if it is on.

If that does not work, start a new thread or go over to discord supabase channel and ask.  Too many people on this thread getting emails with each post...

**Reply by w3b6x9** [reply]
@sidthekid402 please submit a ticket using this form and I can investigate: https://app.supabase.io/support/new
